src/pdm/utils.py

```diff
@@ -453,3 +453,26 @@ def package_installed(package_name: str) -> bool:
         return False
     else:
         return True
+
+
+def validate_project_name(name: str) -> bool:
+    """Check if the project name is validate or not"""
+
+    allowed_chars = set("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-")
```

Comment:
I suppose we could use `string.ascii_letters` and `string.digits` instead:

```suggestion
    allowed_chars = set(string.ascii_letters + string.digits + "_-")
```

Reply:
yeah, you're right, it's much better ~~ anything else